tls-tutorial/doc/tls_in_udfs.md

```diff
@@ -0,0 +1,225 @@
+# TLS in User Defined Functions (UDFs)
+
+## Before you Start
+
+If you haven't already, I recommend you read our ["TLS introduction"](tls_in_udfs.md) to familiarize yourself with the basic concepts of TLS.
+
+You should familiarize yourself with Exasol's [BucketFS](https://docs.exasol.com/db/latest/database_concepts/bucketfs/bucketfs.htm) filesystem.
+
+You will also need:
+
+1. A Java Development Kit (JDK 11 or later, we recommend [OpenJDK](https://openjdk.org/install/))
+2. `keytool` program (comes with the JDK)
+3. [`curl`](https://curl.se/) to access BucketFS and download file from the internet
+
+## UDFs
+
+A User Defined Function (short "UDF") is a way to extend the Exasol database with new features that can be called from within Exasol SQL statements. Basically. A UDF is a program that runs on the Exasol cluster in the context of an SQL query. The UDF gets its parameters from the SQL statement and its return values can be used by other parts of the query.
+
+Each UDF runs in a sandbox that only lives in the context of a single statement. This ensures that UDF can't be used to escape the boundaries of the statement it belongs too and prevents data leakage.
+
+While the sandbox makes the UDF more secure, it unfortunately makes using TLS harder.
+
+## Language Containers
+
+A UDF can be written in an arbitrary programming language but requires an appropriate runtime environment that can be provided as a so-called "Language Container". The Exasol database ships with a standard container providing the runtime environment for Java and Python. Other containers and updates for the standard container are [available for download on GitHub](https://github.com/exasol/script-languages-release/releases).
```

Comment:
"ships" -> "is shipped", though not 100% sure here.

How about R?

Reply:
Changed. You are right, R is contained as well, even if that version is a bit dated. There is a separate container with a newer version, but that is not in the scope of this article.